diff --git a/src/emu/sound.h b/src/emu/sound.h
index 15f6a574..2410bbee 100644
--- a/src/emu/sound.h
+++ b/src/emu/sound.h
@@ -269,7 +269,7 @@ public:
 
 	// copy constructor that sets a different start time
 	read_stream_view(read_stream_view const &src, attotime start) :
-		read_stream_view(src.m_buffer, src.m_buffer->time_to_buffer_index(start, false), src.m_end, src.m_gain)
+		read_stream_view(src.m_buffer, src.m_buffer ? src.m_buffer->time_to_buffer_index(start, false) : src.m_start, src.m_end, src.m_gain)
 	{
 	}
 
